flat_mutation_reader_v2.hh

```diff
@@ -437,6 +437,7 @@ public:
     };
 private:
     std::unique_ptr<impl> _impl;
+    bool _close_required = false;
```

Comment:
If you store _close_required in impl, some of the patch goes away. It makes the handle type lighter weight and  passing it around is cheaper.